src/mpi/coll/allreduce.c

```diff
@@ -303,6 +321,51 @@ int MPIR_Allreduce_intra (
     {
         /* homogeneous */
 
+#ifdef HAVE_EXT_COLL
+        {
+
+            MPIR_Datatype *dt_ptr;
+            MPID_Datatype_get_ptr(datatype, dt_ptr);
+            MPIR_Op *op_p;
+            MPIC_Op_get_ptr(op, &op_p);
+
+            int valid_coll[] = {1,2,3,4};
+            int use_coll = (MPIR_CVAR_USE_ALLREDUCE < 0) ? MPIR_Coll_cycle_algorithm(comm_ptr, valid_coll, 3) : MPIR_CVAR_USE_ALLREDUCE;
+
+            switch (use_coll){
+                case 0:
```

Comment:
That's Ok. This code in MPIR_Allreduce_intra already. In case 0 we jumps to implementation in the same function, right after switch/case statement. That's why we have "goto fn_exit;" in other cases.